plugins/org.eclipse.glsp.server/src/org/eclipse/glsp/server/jsonrpc/DefaultClientSessionManager.java

```diff
@@ -51,17 +48,17 @@ public synchronized boolean connectClient(final GLSPClient client) {
    @Override
    public synchronized boolean createClientSession(final GLSPClient client, final String clientId) {
       connectClient(client);
-      boolean success = clientSessions.putIfAbsent(clientId, client) == null;
+      boolean success = clientSessions.computeIfAbsent(client, c -> new HashSet<>()).add(clientId);
```

Comment:
If you use computeIfAbsent here, we can omit the connectClient(client) call in line 50

Reply:
We still want to notify the session listeners; but we can change the computeIfAbsent with a get()